### ISSUE TITLE
`isabstracttype` seems to be wrong in some instances

### ISSUE THREAD

**Issue by StevenWhitaker** [JuliaLang/julia#48142]
The docstring for `Tuple` states
> Therefore, `Tuple{Any}` is considered an abstract type

However, `isabstracttype(Tuple{Real})` returns `false`.

I get that the docstring for `isabstracttype` says it looks for if a type was declared with the `abstract` keyword, which doesn't apply to `Tuple`, so it's technically not wrong. But it *seems* wrong.

And the problem isn't just for `Tuple`s. For example, `isabstracttype(Array)` also returns `false`, even though `Array` is an abstract type.

So, in summary, even though `isabstracttype` does what the docstring says it does, it *appears* to give wrong information.

**Comment by giordano** [comment]
> For example, `isabstracttype(Array)` also returns `false`, even though `Array` is an abstract type.

What?  No!  `Array` is concrete, but it's also parametric and `Array` alone isn't fully specified, but it's not abstract.

**Comment by vtjnash** [comment]
`Array` is considered abstract, because it cannot be instantiated (non-concrete) but can have subtypes. We do sometimes try to limit our usage of that explanation though, since it distracts users that see it as being solely related to the abstract keyword.

**Comment by giordano** [comment]
I thought non-fully-specified composite parametric types were in a condition were they were both non-instantiatable (so non-"concrete" in a sense) and non-abstract (because it can't be subtyped by a _new_ type), this is also what `isbastracttype` and `isconcretetype` say for `Array`:
```julia
julia> isabstracttype(Array)
false

julia> isconcretetype(Array)
false
```

**Comment by fingolfin** [comment]
Yeah, "concrete" and "abstract" are not opposites.

I think the problem here is the doicstring for `Tuple`. It shouldn't refer to `Tuple{Any}` as an "abstract type", that's just confusing. And it really isn't:
```
julia> isabstracttype(Tuple{Any})
false
```
